utils/sysfs/sysfs.go

```diff
@@ -233,7 +235,20 @@ func (self *realSysFs) GetCacheInfo(id int, name string) (CacheInfo, error) {
 }
 
 func (self *realSysFs) GetSystemUUID() (string, error) {
-	id, err := ioutil.ReadFile(path.Join(dmiDir, "id", "product_uuid"))
+	var uuidPath string
+
+	if strings.Contains(runtime.GOARCH, "ppc64") {
```

Comment:
I'd say the same thing here that @rjnagal suggested. Let's just check all of these and take the first one that provides a UUID